eZ/Publish/Core/Base/Tests/PHPUnit5CompatTrait.php

```diff
@@ -1,53 +0,0 @@
-<?php
-/**
- * File containing PHPUnit 5 Forward Compatibility trait.
- *
- * @copyright Copyright (C) eZ Systems AS. All rights reserved.
- * @license For full copyright and license information view LICENSE file distributed with this source code.
- */
-namespace eZ\Publish\Core\Base\Tests;
-
-/**
- * Trait for PHPUnit 5 Forward Compatibility, for PHPUnit 4.8 use and up.
- */
-trait PHPUnit5CompatTrait
```

Comment:
As discussed we should not remove this, but rather remove createMock on branches where we have it natively, and consider throwing on getMock on all branches so we get rid of getMock usage.

Besides arguments already mentioned (issues for branch merges) it's also used outside of kernel already, one example:
https://github.com/ezsystems/EzSystemsRecommendationBundle/commit/128427239b2392938834f0f099a2011878d31ac2#diff-e4b181b287b96f000150d27b23964aebR9